packages/widget/src/hooks/useAddressAndENSValidation.ts

```diff
@@ -0,0 +1,79 @@
+import { useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useConfig } from 'wagmi';
+import { ChainType } from '@lifi/sdk';
+import { getEnsAddress } from '@wagmi/core';
+import { useToAddressRequirements } from '../hooks';
+import { normalize } from 'viem/ens';
+import { useFieldActions } from '../stores';
+import type { AddressType } from '../stores';
+import { getChainTypeFromAddress } from '../utils';
+
+type ValidResponse = {
+  isValid: true;
+  error: string;
+  chainType: ChainType;
+  addressType: AddressType;
+};
+
+type InvalidResponse = {
+  isValid: false;
+  error: string;
+};
+export const useAddressAndENSValidation = () => {
```

Comment:
Let's rename it to `useAddressValidation` because I just thought we could add more services like ENS in the future and it is not something standard, it's one of many